src/ResourceGenerator.php

```diff
@@ -123,7 +123,7 @@ public function fromArray(array $data, string $uri = null) : HalResource
      *     against types registered in the metadata map.
      * @param ServerRequestInterface $request
      */
-    public function fromObject($instance, ServerRequestInterface $request) : HalResource
+    public function fromObject($instance, ServerRequestInterface $request, int $depth = 0) : HalResource
```

Comment:
To help me as I create the changelog: could you please collate everywhere there is an additional optional argument passed to a public method? If I have that list, it will be far easier to communicate to users what may have changed in any extensions or custom implementations they have created.

Thanks in advance!

Reply:
- [Metadata\RouteBasedResourceMetadata::__construct](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-7a2468397c35eb8f00f6b4f1402091f3R34)
- [ResourceGenerator::fromObject](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-2577e1c736a62f8dcb0285047e60c6dfR126)
- [ResourceGenerator\StrategyInterface::createResource](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-80d6dfcf240ad1625d0271caf37c44dcR27)
- [ResourceGenerator\RouteBasedCollectionStrategy::createResource](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-a45d3ac0928e28765228eae130159a64R29)
- [ResourceGenerator\RouteBasedResourceStrategy::createResource](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-a45d3ac0928e28765228eae130159a64R29)
- [ResourceGenerator\UrlBasedCollectionStrategy::createResource](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-7b3ab3eefd5973646678c936f1c22043R37)
- [ResourceGenerator\UrlBasedResourceStrategy::createResource](https://github.com/zendframework/zend-expressive-hal/pull/57/files#diff-9de76abaa48b255ffea38751d58b9af6R25)